File: acmicpc.net/1010/Main.java
import java.math.BigInteger;
import java.util.Scanner;

public class Main
{
	public static void main(String... args)
	{
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for(int i = 0; i < T; i++)
		{
			System.out.println(solve(sc.nextInt(), sc.nextInt()).toString());
		}
	}

	public static BigInteger solve(int s, int b)
	{
		BigInteger result = BigInteger.ONE;
		for(int i = 0; i < s; i++)
		{
			result = result.multiply(new BigInteger(Integer.toString(b - i)));
		}
		for(int i = 0; i < s; i++)
		{
			result = result.divide(new BigInteger(Integer.toString(i + 1)));
		}
		return result;
	}
}

File: acmicpc.net/1010/main.js
var sc = new java.util.Scanner(java.lang.System["in"]);

function solve(s, b)
{
	var result = java.math.BigInteger.ONE;
	for(var i = 0; i < s; i++)
	{
		result = result.multiply(new java.math.BigInteger(b - i));
	}
	for(var i = 0; i < s; i++)
	{
		result = result.divide(new java.math.BigInteger(i + 1));
	}
	return result;
}

var T = sc.nextInt();
for(var i = 0; i < T; i++)
{
	java.lang.System.out.println(solve(sc.nextInt(), sc.nextInt()).toString());
}
